nats/pkg/server/nats/nats.go

```diff
@@ -4,64 +4,56 @@ import (
 	"context"
 	"time"
 
-	natsServer "github.com/nats-io/nats-server/v2/server"
-	stanServer "github.com/nats-io/nats-streaming-server/server"
+	nserver "github.com/nats-io/nats-server/v2/server"
 )
 
 var NATSListenAndServeLoopTimer = 1 * time.Second
 
 type NATSServer struct {
-	ctx context.Context
-
-	natsOpts *natsServer.Options
-	stanOpts *stanServer.Options
-
-	server *stanServer.StanServer
+	ctx             context.Context
+	jetStreamConfig *nserver.JetStreamConfig
+	server          *nserver.Server
 }
 
-// NewNATSServer returns a new NATSServer
-func NewNATSServer(ctx context.Context, opts ...Option) (*NATSServer, error) {
+func NewNATSServer(ctx context.Context, logger nserver.Logger, opts ...Option) (*NATSServer, error) {
+	options := &nserver.Options{}
 
-	server := &NATSServer{
-		ctx:      ctx,
-		natsOpts: &stanServer.DefaultNatsServerOptions,
-		stanOpts: stanServer.GetDefaultOptions(),
+	for _, o := range opts {
+		o(options)
 	}
 
-	for _, o := range opts {
-		o(server.natsOpts, server.stanOpts)
+	server, err := nserver.NewServer(
+		options,
+	)
+	if err != nil {
+		return nil, err
 	}
 
-	return server, nil
+	server.SetLoggerV2(logger, true, true, false)
+
+	c := &nserver.JetStreamConfig{
+		StoreDir: "/tmp/ocis-jetstream", // TODO: configurable
```

Comment:
Is this a future TODO or a forgotten one?

Reply:
Thanks, I made it configurable now